pmd-core/src/main/java/net/sourceforge/pmd/properties/PropertyDescriptorUtil.java

```diff
@@ -23,29 +23,29 @@
     static {
         Map<String, PropertyDescriptorExternalBuilder<?>> temp = new HashMap<>(18);
         temp.put("Boolean", BooleanProperty.extractor());
-        temp.put("List<Boolean>", BooleanMultiProperty.extractor());
+        temp.put("List[Boolean]", BooleanMultiProperty.extractor());
```

Comment:
Wouldn't that break compatibility with existing (Java-based) rules, that use already multi-valued properties? Maybe we can have both variants in the map?

Reply:
Java-based rules don't use these factories, only XPath rules.

It breaks compatibility, but no more than the `List<>` form. As of [release 5.8.1](https://github.com/pmd/pmd/blob/pmd_releases/5.8.1/pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/properties/factories/PropertyDescriptorUtil.java#L57), the name of those properties had the form  `Boolean[]`, like an array (the `List<>` form was introduced during #504). Whatever we choose breaks compatibility. 

But I have to stress that none of that is documented behaviour for now. It's very unlikely anyone uses it, and if they do, documentation and release notes may be enough.

Reply:
@oowekyala I agree - the old code for XPath 1.0 (Jaxen) just does a toString (which was for multi-valued properties just an array) and with XPath 1.0 compatible/2.0 (Saxon) we rejected such properties.

So, yes, we can choose freely here. Avoiding the `<` helps for usage inside XML, so I'm good with your suggestion `List[Boolean]`.